README.md

```diff
@@ -8,3 +8,33 @@ Currently implemented applications are
     - Full-service cross-chain message delivery application that is configurable to listen to specific source and destination chain pairs and relay messages according to its configured rules.
 2. [Signature Aggregator](signature-aggregator/README.md)
     - Lightweight API that requests and aggregates signatures from validators for any ICM message, and returns a valid signed message that the user can then self-deliver to the intended destination chain.
+
+## Updating dependencies and E2E testing
+
+Applications in this repository depend on the following upstream repositories, both directly in terms of code imports defined in `go.mod` file as well as indirectly for E2E tests where binary versions are used to spin up the test network via `tmpnet`:
+
+1.  [avalanchego](https://github.com/ava-labs/avalanchego/)
+2.  [coreth](https://github.com/ava-labs/coreth) (indirectly)
+3.  [subnet-evm](https://github.com/ava-labs/subnet-evm)
+
+> [!NOTE]
+> We require any commits referenced in our `main` branch to be present in the default branches of the repositories above, but during active development it might be useful to work against changes in progress that are still on feature branches.
+
+When developing such features that require updates to one or more of the above, care must be taken to understand where the relevant code comes from. The binaries of applications built in this repo are built against versions referenced in the `go.mod` file. The E2E tests run against a simulated network running locally that is started by calling a separately compiled `avalanchego` binary as well as its plugins. These are compiled based on the values of `AVALANCHEGO_VERSION` in the local checkout of `subnet-evm` when running the tests locally and directly in this repository's `./scripts/versions.sh` when running E2E tests remotely through GitHub actions. 
```

Comment:
This is currently only in a branch, but we equate the versions in `go.mod` and the binary installed for the e2e tests like so in `teleporter`: https://github.com/ava-labs/teleporter/blob/staking-contract/scripts/versions.sh#L32

We could simplify this documentation by incorporating this approach here.

Reply:
Yeah but it's possible and valid to do both. At the very least this explains which code comes from where.

Reply:
I.e. it might be useful to test a version of relayer code against a tmpnet running a separate version of the code. I agree that we should make the `extract-commit` change in this repo as well, but for development and troubleshooting I still want some version of the documentation above.